### ISSUE TITLE
Can't make a defstruct with name population

### ISSUE THREAD

**Issue by Slids** [Clozure/ccl#324]
First:
(defpackage foo)
Then:
(in-package 'foo)
Finally:
(defstruct population) 
Error:
Cannot redefine built-in type POPULATION
 
This seems bizarre. Maybe I'm to use to sbcl but I would expect it to make a type in the foo namespace with name population. Seems like this is a obviously named struct and it probably shouldn't be forbidden...

**Comment by xrme** [comment]
On CCL, the default :use list includes the :ccl package.  When you don't specify :use, the default is implementation dependent. Since your foo package uses ccl, you are getting a conflict with ccl:population.

Specify the :use list explicitly, and you won't get this conflict.

```
(defpackage foo
 (:use cl))
(in-package :foo)
(defstruct population)
````

**Comment by Slids** [comment]
Sure, it's fine by the standard but why would you default to having :ccl in your use call?
It's more burden on users, especially who want portable code.

**Comment by phoe** [comment]
Users who want code that behaves the same on all implementations shouldn't depend on implementation-defined functionality. The `:use` list is such a case - it is implementation-defined unless it is explicit.

Let's consider `(defpackage foo) (mapcar #'package-name (package-use-list (find-package 'foo)))`. For SBCL and ABCL, the result is empty. ECL, CLISP, ACL, CMUCL have `CL` only, while Lispworks has `CL`, `HCL`, and `LISPWORKS`. CCL doesn't really stick out with its default use list of `CL` and `CCL`.

**Comment by Slids** [comment]
But why would you add it?
Whats so great in the ccl package that it's worth making my package that much larger with more symbols, without my explicit say so?

**Comment by xrme** [comment]
Long-standing implementation behavior is to include the ccl package in the default package use list. If CCL were to be re-written today, maybe a different choice would be made. I don't know the reason why it was decided to include the CCL package in the default list, but one guess might be that the CCL package originally contained all the Macintosh-specific stuff, and it was thought that most users of Coral (later Macintosh) Common Lisp would want to use that.

CCL isn't trying to trick you or go behind your back in nefarious scheme to bloat your own stuff. It is totally fine and reasonable for you not to want to use the packages in the implementation-defined default list. Write out the :use option to defpackage, and you will get exactly what you want.

TL;DR: historical reasons

**Comment by Slids** [comment]
Never said ccl was trying to trick me, just disagree with the choice.
I do agree that due to historical reasons we shouldn't change it,
but I can still complain!
